plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,575 @@
+        callback = function()
+            G_reader_settings:flipNilOrFalse("press_key_does_hotkeyshortcuts")
+            UIManager:askForRestart()
+        end,
+    }
+    menu_items.hotkeyshortcuts = {
+        sorting_hint = "physical_buttons_setup",
+        text = _("Keyboard shortcuts"),
+        sub_item_table = {
+            {
+                text = _("Cursor keys"),
+                sub_item_table = self:genSubItemTable(cursor_keys),
+            },
+            {
+                text = _("Page-turn buttons"),
+                enabled_func = function() return self.hotkey_mode == "hotkeyshortcuts_reader" end,
+                sub_item_table = self:genSubItemTable(pg_turn),
+            },
+            {
+                text = _("Function keys"),
+                sub_item_table = self:genSubItemTable(fn_keys),
+            },
+        },
+    }
+
+    if Device:hasKeyboard() then
+        table.insert(menu_items.hotkeyshortcuts.sub_item_table, {
+            text = _("Alphabet keys"),
+            sub_item_table = self:genSubItemTable({
+                "alt_plus_a", "alt_plus_b", "alt_plus_c", "alt_plus_d", "alt_plus_e", "alt_plus_f", "alt_plus_g", "alt_plus_h", "alt_plus_i",
+                "alt_plus_j", "alt_plus_k", "alt_plus_l", "alt_plus_m", "alt_plus_n", "alt_plus_o", "alt_plus_p", "alt_plus_q", "alt_plus_r",
+                "alt_plus_s", "alt_plus_t", "alt_plus_u", "alt_plus_v", "alt_plus_w", "alt_plus_x", "alt_plus_y", "alt_plus_z",
+            }),
+        })
+    end
+end
+
+--[[
+    Description:
+    This function overrides existing `registerKeyEvents()` functions in various modules to resolve conflicts and customize key event handling.
+    It modifies the key event registration for several reader and file manager modules based on device capabilities and user settings.
+
+    Modules and their modifications:
+    - ReaderBookmark: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderConfig: Customizes `ShowConfigMenu` key event based on user settings.
+    - ReaderDictionary: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderLink: Customizes `GotoSelectedPageLink` key event for devices with screen keyboard or symbol key.
+    - ReaderSearch: Customizes `ShowFulltextSearchInputBlank` key event for devices with a keyboard.
+    - ReaderToc: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderThumbnail: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderWikipedia: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderUI: Customizes `Home` and `KeyContentSelection` key events based on device capabilities.
+    - FileManager: Customizes `Home` and `Back` key events, and conditionally removes `Close` key event.
+    - FileSearcher: Customizes `ShowFileSearchBlank` key event for devices with keyboards.
+    - FileManagerMenu: Customizes `ShowMenu` key event for devices with keys.
+]]
+function HotKeyShortcuts:overrideConflictingFunctions()
```

Comment:
Since this plugin appears to be actually handling key input, these `registerKeyEvents` in all the modules should be removed and folded in to the plugin, much less of a maintenance headache. Similar to how many of the module gestures were moves to the plugin. As @poire-z  said nilling them to register the same ones in the plugin doesn't make much sense

Reply:
I actually don't mind that. I'd rather keep having the default key behaviour in each core module, and not have to care about this plugin (which may be disabled - and NT users could be left with no key working at all :)).

Reply:
Gamepads, BT page turners and the like remain "new" devices from the code's perspective that have always been planned to be combined with any work on keyboard shortcuts how @yparitcher stated. That remark was remarkably successful at introducing worry where there was essentially none. ;-)